### PR TITLE
vte: fix build on darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

As in the title.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/vte/default.nix [hunk 1]
```diff
@@ -92,8 +92,13 @@ stdenv.mkDerivation rec {
     "-D_b_symbolic_functions=false"
   ];
 
-  # error: argument unused during compilation: '-pie' [-Werror,-Wunused-command-line-argument]
-  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isMusl "-Wno-unused-command-line-argument";
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.hostPlatform.isMusl [
```
Comment (jtojnar) [inline pkgs/development/libraries/vte/default.nix:95]: This variable should be a string, see https://github.com/NixOS/nixpkgs/pull/72074

Comment (wegank) [reply pkgs/development/libraries/vte/default.nix]: Done. Thanks!

**Comment by jtojnar** [comment]
Also is there an upstream bug – I would expect Homebrew to suffer from this as well but with a quick search I only noticed https://gitlab.gnome.org/GNOME/vte/-/issues/2592.

**Comment by wegank** [comment]
I think it is the same issue: now switching to `-D_DARWIN_C_SOURCE`.

**Comment by jtojnar** [comment]
Might be nicer to switch to https://gitlab.gnome.org/GNOME/vte/-/commit/1f1f177ff797ac2bb453168951135865cfded900.patch then. This would allow us to easily see when it is no longer necessary.

**Review by jtojnar — APPROVED** [review]

**Comment by jtojnar** [comment]
Thanks.

**Pull request merged**